src/firefly/js/visualize/ui/ImageMetaDataToolbarView.jsx

```diff
@@ -93,20 +101,40 @@ export function ImageMetaDataToolbarView({activePlotId, viewerId, viewerPlotIds,
                 }
             </div>
             {showPager && <ImagePager pageSize={10} tbl_id={activeTable.tbl_id} />}
+            {handleInlineTools && makeInlineRightToolbar(vr,pv,pvDlAry)}
         </div>
     );
 }
 
 ImageMetaDataToolbarView.propTypes= {
+    dlAry : PropTypes.arrayOf(React.PropTypes.object),
     activePlotId : PropTypes.string,
     viewerId : PropTypes.string.isRequired,
     layoutType : PropTypes.string.isRequired,
     viewerPlotIds : PropTypes.arrayOf(PropTypes.string).isRequired,
     activeTable: PropTypes.object,
-    converterFactory: PropTypes.func
+    converterFactory: PropTypes.func,
+    handleInlineTools : PropTypes.bool
 };
 
 
+function makeInlineRightToolbar(visRoot,pv,dlAry){
+    if (!pv) return false;
+
+    var lVis= BrowserInfo.isTouchInput() || visRoot.apiToolsView;
+    var tb= visRoot.apiToolsView;
+    return (
+        <div>
+            <VisInlineToolbarView
+                plotId={pv.plotId} dlAry={dlAry}
+                showLayer={lVis}
+                showExpand={true}
+                showToolbarButton={tb}
+                showDelete ={false}
+            />
+        </div>
+    );
+}
```

Comment:
This should be a functional component.. You never know.  May get that 'performance' gain some day.